contracts/libraries/VaultLib.sol

```diff
@@ -278,14 +278,20 @@ library VaultLibrary {
         return self.vault.userLvBalance[user].balance;
     }
 
+    // returns the amount of PA as per price of PA with LV and given LV amount
+    // lv price = paAvailable/lvTotalSupply
+    function _calculatePaPriceForLv(State storage self, uint256 lvAmt) internal view returns (uint256 paAmount) {
+        return lvAmt * ERC20(self.info.pair0).balanceOf(address(this)) / ERC20(self.vault.lv._address).totalSupply();
+    }
+
```

Comment:
use the pa balance, not the actual balance. since there could be many vault with the sam PA but with different RA